### PR TITLE
Added isort, ran isort on all files

### PR DESCRIPTION
**Pull request by akash-suresh**
Resolves https://github.com/Axelrod-Python/Axelrod/issues/1326

@drvinceknight I have added an `isort` config, and run `isort` over all the files.

Let me know if there are any modifications needed in the `.isort.cfg` file.

### CODE REVIEW

**Comment by drvinceknight** [comment]
The CI has failed with the following traceback:

```
************* Module axelrod.strategies._strategies
29
axelrod/strategies/_strategies.py:81:0: W0611: Unused Cycler imported from cycler (unused-import)
30
axelrod/strategies/_strategies.py:81:0: W0611: Unused EvolvableCycler imported from cycler (unused-import)
31
axelrod/strategies/_strategies.py:118:0: W0611: Unused Gambler imported from gambler (unused-import)
32
axelrod/strategies/_strategies.py:153:0: W0611: Unused HMMPlayer imported from hmm (unused-import)
33
axelrod/strategies/_strategies.py:166:0: W0611: Unused LookerUp imported from lookerup (unused-import)
```

Let me know if you'd like me to take a closer look and figure out what's going on.

**Comment by akash-suresh** [comment]
@drvinceknight I have fixed the issues. 😄

File: axelrod/strategies/__init__.py [hunk 1]
```diff
@@ -99,18 +100,14 @@ def filtered_strategies(filterset, strategies=all_strategies):
     Applies the filters defined in the given filterset dict and returns those
     strategy classes which pass all of those filters from the given list of
     strategies.
-
```
Comment (gaffney2010) [inline axelrod/strategies/__init__.py:102]: Why delete these lines?

Comment (akash-suresh) [reply axelrod/strategies/__init__.py]: It probably got introduced, when I copied the raw file from git to get the dependencies back in the same order. 
Fixed it now. 👍

**Comment by drvinceknight** [comment]
This is great! 

It would be nice to add a check to the CI so that imports don't get "unsorted" in the future.

In https://github.com/Axelrod-Python/Axelrod/blob/master/.github/workflows/config.yml could you add:

```
    - name: Check imports are sorted
      run: |
        python -m pip install isort
        python -m isort --check-only
```

**Comment by drvinceknight** [comment]
Note that when/if we get to https://github.com/Axelrod-Python/Axelrod/issues/1312 it's possible that the `isort` and `black` checks contradict each other (I mentioned this at https://github.com/Axelrod-Python/Axelrod/issues/1326) but we'll cross that bridge if/when we need to and for now it would be good to have a check for isort :+1:

**Comment by akash-suresh** [comment]
> This is great!
> 
> It would be nice to add a check to the CI so that imports don't get "unsorted" in the future.
> 
> In https://github.com/Axelrod-Python/Axelrod/blob/master/.github/workflows/config.yml could you add:
> 
> ```
>     - name: Check imports are sorted
>       run: |
>         python -m pip install isort
>         python -m isort --check-only
> ```

Done! 😄

**Comment by drvinceknight** [comment]
Fantastic, thanks for the contribution @akash-suresh :+1:

**Pull request merged**